### ISSUE TITLE
The specified metadata type is unsupported: [expressionsetobjectalias]

### ISSUE THREAD

**Issue by TonyWhiteSMS** [forcedotcom/cli#2589]
### Summary
Error attempting retrieve the metadata for the Business Rules Engine related objects

1. Sign-up for a Business Rules Engine Trailhead Org - https://trailhead.salesforce.com/promo/orgs/business-rules-engine-
2. Connect CLI to the Org
3. Setup project
4. Retrieve the Expression Set Object Aliases
`sfdx force:source:retrieve -m ExpressionSetObjectAlias
`

### Expected result
Metadata for the Expression Set objects should be downloaded as per https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_expressionsetobjectalias.htm

### Actual result
```
$ sfdx force:source:retrieve -m ExpressionSetObjectAlias
Warning: We plan to deprecate this command in the future. Try using the "project retrieve start" command instead.
Preparing retrieve request... Error
Error (1): The specified metadata type is unsupported: [expressionsetobjectalias]
```

### System Information
```json
{
  "architecture": "darwin-x64",
  "cliVersion": "@salesforce/cli/2.19.8",
  "nodeVersion": "node-v21.2.0",
  "osVersion": "Darwin 22.6.0",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "shell": "zsh",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.2 (core)",
    "@oclif/plugin-commands 3.0.7 (core)",
    "@oclif/plugin-help 6.0.7 (core)",
    "@oclif/plugin-not-found 3.0.4 (core)",
    "@oclif/plugin-plugins 4.1.8 (core)",
    "@oclif/plugin-search 1.0.8 (core)",
    "@oclif/plugin-update 4.1.3 (core)",
    "@oclif/plugin-version 2.0.7 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.3 (core)",
    "@oclif/plugin-which 3.0.10 (core)",
    "@salesforce/cli 2.19.8 (core)",
    "apex 3.0.6 (core)",
    "auth 3.0.7 (core)",
    "data 3.0.1 (core)",
    "deploy-retrieve 2.1.3 (core)",
    "info 3.0.3 (core)",
    "limits 3.0.2 (core)",
    "marketplace 1.0.6 (core)",
    "org 3.0.7 (core)",
    "packaging 1.27.2 (core)",
    "schema 3.0.4 (core)",
    "settings 2.0.6 (core)",
    "sobject 1.0.2 (core)",
    "source 2.11.4 (core)",
    "telemetry 3.1.2 (core)",
    "templates 56.0.2 (core)",
    "trust 3.0.4 (core)",
    "user 3.0.4 (core)",
    "sfdmu 4.30.0 (user)"
  ]
}
```

**Comment by shetzel** [comment]
The fix is now in the sf CLI release candidate v2.24.4.

**Comment by shetzel** [comment]
The fix is now in the production sf CLI, v2.24.4

**Comment by shetzel** [comment]
The fix is now in the production sf CLI, v2.24.4
